### PR TITLE
Interval improvement

### PR DESCRIPTION
**Pull request by BiPhan4**
@TheMarstonConnell Still a work in progress. Requested your review so you can follow along. The objective is to prevent providers from committing proofs at the same time. Continuously changing the sleep interval between committing proofs is an attempt at this.

The sleep interval is a random number between 30 and 120 seconds. Given the 1 hour proving window, a maximum sleep interval of 2 minutes is a safe estimate of how often a provider needs to commit proofs. As storage demand grows, the maximum sleep time could be decreased. 

We still give providers the option of setting a custom sleep interval, if they are confident that their machine will commit all their proofs with this custom interval.

### CODE REVIEW

File: jprov/jprovd/provider_commands.go [hunk 1]
```diff
@@ -34,12 +34,12 @@ func StartServerCommand() *cobra.Command {
...
 	cmd.Flags().Bool(types.HaltStraysFlag, false, "Debug flag to stop picking up strays.")
-	cmd.Flags().Uint16(types.FlagInterval, 32, "The interval in seconds for which to check proofs.")
+	cmd.Flags().Uint16(types.FlagInterval, 32, "The interval in seconds for which to check proofs. Must be >=120 if you need a custom interval")
 	cmd.Flags().Uint(types.FlagThreads, 10, "The amount of stray threads.")
 	cmd.Flags().Int(types.FlagMaxMisses, 16, "The amount of intervals a provider can miss their proofs before removing a file.")
-	cmd.Flags().Int64(types.FlagChunkSize, 10240, "The size of a single chunk.")
+	cmd.Flags().Int64(types.FlagChunkSize, 10240, "The size of a single file chunk.")
 	cmd.Flags().Int64(types.FlagStrayInterval, 20, "The interval in seconds to check for new strays.")
-	cmd.Flags().Int(types.FlagMessageSize, 500_000, "The max message size in bytes to submit to the chain at one time.")
+	cmd.Flags().Int(types.FlagAllMessagesSize, 500_000, "The max size of all messages in bytes to submit to the chain at one time.")
```
Comment (TheMarstonConnell) [inline jprov/jprovd/provider_commands.go:42]: We need to properly deprecate this by keeping both and ignoring `FlagAllMessageSize` if `FlagMessageSize` is set or else we will break some service files.

File: jprov/server/proofs.go [hunk 2]
```diff
@@ -169,6 +170,15 @@ func postProofs(cmd *cobra.Command, db *leveldb.DB, q *queue.UploadQueue, ctx *u
...
 	for {
+		interval := intervalFromCMD
+
+		if interval < 120 { // If the provider picked an interval that's less than 2 minutes, we generate a random interval for them anyways
+
+			r := rand.New(rand.NewSource(time.Now().UnixNano()))
+			interval = uint16(r.Intn(91) + 30)
+
+		}
+		ctx.Logger.Info(fmt.Sprintf("The interval between proofs is now %d", interval))
```
Comment (TheMarstonConnell) [inline jprov/server/proofs.go:181]: Change to debug log

File: jprov/server/proofs.go [hunk 3]
```diff
@@ -169,6 +170,15 @@ func postProofs(cmd *cobra.Command, db *leveldb.DB, q *queue.UploadQueue, ctx *u
 	}
 
 	for {
+		interval := intervalFromCMD
+
+		if interval < 120 { // If the provider picked an interval that's less than 2 minutes, we generate a random interval for them anyways
```
Comment (TheMarstonConnell) [inline jprov/server/proofs.go:175]: 2 minutes is a pretty small window, this'll most likely still cause conflicts. I personally have my provider set to 45 minutes. Generating an interval of say 20-40 minutes seems like a more beneficial arrangement.

**Review by TheMarstonConnell — CHANGES_REQUESTED** [review]

File: jprov/jprovd/provider_commands.go (hunk 1 above)
Comment (BiPhan4) [reply jprov/jprovd/provider_commands.go]: Understood. Reverting back for now.

File: jprov/server/proofs.go (hunk 2 above)
Comment (BiPhan4) [reply jprov/server/proofs.go]: done

**Review by TheMarstonConnell — APPROVED** [review]

**Pull request merged**